### PR TITLE
Adds Farmer Boy's Spider

### PR DESCRIPTION
**Pull request by stephinson**
Scrapes 100 [locations](https://www.farmerboys.com/locations/)

Closes https://github.com/alltheplaces/alltheplaces/issues/3530

Here's the sample output:

```json
{"type": "Feature", "id": "8UiK4I64urhrJQSZF3xz5ySVQxE=", "properties": {"ref": "baldwin-park-merced-ave", "@spider": "farmerboys", "addr:street": "14430 Merced Ave.", "addr:city": "Baldwin Park", "addr:state": "CA", "addr:postcode": "91706", "addr:country": "US", "name": "Baldwin Park - Merced Ave.", "phone": "(626) 856-1000", "website": "https://www.farmerboys.com/locations/location-detail.php?loc=baldwin-park-merced-ave", "opening_hours": "Mon-Sun  6:00a-12:00a", "brand": "Farmer Boy's"}, "geometry": {"type": "Point", "coordinates": [-117.962104, 34.072683]}},
{"type": "Feature", "id": "JBXNafOxNrkP615zMzgBWMSxWsA=", "properties": {"ref": "beaumont-2nd-street-marketplace", "@spider": "farmerboys", "addr:street": "1538 2nd St. Marketplace", "addr:city": "Beaumont", "addr:state": "CA", "addr:postcode": "92223", "addr:country": "US", "name": "Beaumont - 2nd St. Marketplace", "phone": "(951) 845-4414", "website": "https://www.farmerboys.com/locations/location-detail.php?loc=beaumont-2nd-street-marketplace", "opening_hours": "Mon-Sat 6:00a-9:00p Sun 7:00a-9:00p", "brand": "Farmer Boy's"}, "geometry": {"type": "Point", "coordinates": [-116.956538, 33.923217]}},
{"type": "Feature", "id": "Ucq0OkK9h3xcmGxDsOuSN0W7AaU=", "properties": {"ref": "bloomington-south-riverside-ave", "@spider": "farmerboys", "addr:street": "3356 S. Riverside Ave.", "addr:city": "Bloomington", "addr:state": "CA", "addr:postcode": "92316", "addr:country": "US", "name": "Bloomington - S. Riverside Ave.", "phone": "(951) 248-1244", "website": "https://www.farmerboys.com/locations/location-detail.php?loc=bloomington-south-riverside-ave", "opening_hours": "Mon-Sat 6:00a-10:00p Sun 7:00a-10:00p", "brand": "Farmer Boy's"}, "geometry": {"type": "Point", "coordinates": [-117.367757, 34.03887]}},
{"type": "Feature", "id": "IgzQqkNxeVTc_zDP5bB4c8uPrA8=", "properties": {"ref": "bloomington-valley-blvd", "@spider": "farmerboys", "addr:street": "18694 Valley Blvd.", "addr:city": "Bloomington", "addr:state": "CA", "addr:postcode": "92316", "addr:country": "US", "name": "Bloomington - Valley Blvd.", "phone": "(909) 873-5800", "website": "https://www.farmerboys.com/locations/location-detail.php?loc=bloomington-valley-blvd", "opening_hours": "Mon-Sat 6:00a-10:00p Sun 7:00a-10:00p", "brand": "Farmer Boy's"}, "geometry": {"type": "Point", "coordinates": [-117.396634, 34.070685]}},
{"type": "Feature", "id": "bQFPJxWsLBkfKF40hot1fASpjJI=", "properties": {"ref": "brea-lambert-rd", "@spider": "farmerboys", "addr:street": "500 W. Lambert Rd.", "addr:city": "Brea", "addr:state": "CA", "addr:postcode": "92821", "addr:country": "US", "name": "Brea - W. Lambert Rd.", "phone": "(714) 990-2424", "website": "https://www.farmerboys.com/locations/location-detail.php?loc=brea-lambert-rd", "opening_hours": "Mon-Thur 6:00a-9:00p Fri-Sat 6:00a-10:00p Sun 7:00a-9:00p", "brand": "Farmer Boy's"}, "geometry": {"type": "Point", "coordinates": [-117.902041, 33.924327]}}

```

### CODE REVIEW

File: locations/spiders/farmerboys.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import re
+
+import scrapy
+from locations.items import GeojsonPointItem
+from scrapy.selector import Selector
+
+
+class FarmerBoys(scrapy.Spider):
+    name = "farmerboys"
+    item_attributes = {"brand": "Farmer Boy's"}
```
Comment (Cj-Malone) [inline locations/spiders/farmerboys.py:12]:
```patch
-item_attributes = {"brand": "Farmer Boy's"}
+item_attributes = {"brand": "Farmer Boys", "brand_wikidata": "Q5435711"}
```

**Review by Cj-Malone — CHANGES_REQUESTED** [review]
This looks good, a few comments though.

**Comment by Cj-Malone** [comment]
It also looks like the image is available, as "https://www.farmerboys.com/images/locations/" + location_pic, it's not a requirement, but a nice to have.

File: locations/spiders/farmerboys.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+                "postcode": location["postal_code"],
+                "state": location["state"],
+                "country": "US",
+                "phone": location["phone"],
+                "website": "https://www.farmerboys.com/locations/location-detail.php?loc="
+                + location["location_url"],
+                "lat": float(location["lat"]) if location["lat"] else None,
+                "lon": float(location["lng"]) if location["lng"] else None,
+                "opening_hours": " ".join(
+                    Selector(text=location["location_hours"]).xpath("//text()").getall()
```
[the remark this answers is not in the document]
Comment (stephinson) [reply locations/spiders/farmerboys.py:41]: That would've worked if all opening hours followed a pattern, but that's not the case

`Mon-Sun Dining Room  6:00a-10:00p Drive Thru 6:00a-11:00p`

**Review by Cj-Malone — APPROVED** [review]
IMO less than 100% coverage, but parsable opening hours is better than 100% of opening hours that aren't parsable. But I'm happy to clean this up in another PR.

This looks good to go, thank you.

**Pull request merged**